### DISCUSSION TITLE
Installation Issues: Help for New Users and Developers (Continuously Updated)

### DISCUSSION THREAD

**Discussion by yuki-2025** [camel-ai/camel#466] [Ideas]
Just sharing some information with new users who are interested in contributing to and using camel-ai. At the same time, I hope any expert can help address the issues described below.

Bug raised: https://github.com/camel-ai/camel/issues/465 

To differentiate between a normal user and a developer user, I refer to the normal user as '**User**' and the developers of our camel project as '**Developer**'. The main issue concerns dependency problems; strict environments might affect both user and developer experiences. 

I followed the instructions in the wiki or readme and tested the installation on two different PCs, and encountered the problems described below, some of them have solutions: (Note: The use of an ipynb file is just to facilitate showing the problem and is not related to actual development.)

 **For Beginner User**
The installation instructions as a user:
https://github.com/camel-ai/camel/wiki/Installation-and-Setup and https://github.com/camel-ai/camel
![image](https://github.com/camel-ai/camel/assets/159591455/e2dc7edd-c84b-48b7-8ac5-bfc8cdba60c1)

1. pip does not work with Python 3.12 and above.
![image](https://github.com/camel-ai/camel/assets/159591455/2c5f5eb0-6b25-41f1-b01c-23fd501851c9)
Solution: Install the Python 3.11 version of the kernel; it is not a significant problem.

2. After executing 'pip install all', normal users are still unable to use the library; they cannot run 'role_playing.py'. It seems that pip alone might not be sufficient for normal users. Executing pip install 'camel-ai[all]' results in the same problem. (ModuleNotFoundError: No module named 'docstring_parser')
![image](https://github.com/camel-ai/camel/assets/159591455/f93e79a0-128c-4cff-9a72-a5637264738a)
  #### $\textcolor{red}{\textsf{Solution: So far, I have not been able to find an alternative. I believe the only way forward is to install from the source.}}$ 

-----------

**For Beginner Developer**
The instructions to install as a developer. 
https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md
```
# Clone github repo
git clone https://github.com/camel-ai/camel.git

# Change directory into project directory
cd camel

# Activate camel virtual environment
poetry shell

# Install camel from source
# It takes about 75s to resolve dependencies and 10s to install them, depending on your hardware and network,
poetry install --with dev,docs

# Or if you want to use all other extra packages
poetry install --with dev,docs -E all  # (Optional)

# The following command installs a pre-commit hook into the local git repo,
# so every commit gets auto-formatted and linted.
pre-commit install

# Run camel's unit tests
pytest test

# Exit the virtual environment
exit
```

1. You might encounter a poetry problem after performing a git clone.
![image](https://github.com/camel-ai/camel/assets/159591455/89b78788-87b6-47f1-b808-60261f7053f1)
Solution: For Windows users, you cannot install your Python kernel via the Microsoft App Store, and installing Conda will not resolve the issue either. Simply installing Python 3.11 directly from python.org is sufficient.

2. You might encounter a Torch library error when executing code, even after successfully initiating a Poetry shell.
![image](https://github.com/camel-ai/camel/assets/159591455/12d91243-ef39-4117-8157-f04b3402896c)
Solution Attempted: Despite running pip install torch, the problem persists.

3. You might encounter errors in pytest after successfully executing poetry install and pre-commit install.
![image](https://github.com/camel-ai/camel/assets/159591455/a7ff6e86-f7ac-4279-9322-6f19ceb100dd)
 #### $\textcolor{red}{\textsf{Solution: I tried to installed qdrant but same errors. At this point, I am not sure how to resolve this issue.}}$

**Comment by Appointat** [comment]
@yuki-2025  Thank you for your feedback. However, I believe the issue with poetry that you mentioned is due to a problem with your local environment. You can use ChatGPT to help you address issues related to poetry. And, Pytest is a testing tool that generally runs in the GitHub action environment, which differs from your local env; therefore, errors can occur in the local env.

**Comment by ashwin296** [comment]
Not sure if this is solved yet but it looks like for the first 2, the error message indicates that `pip install docstring_parser` should work.
